site/content/api/Tenant-API.md

```diff
@@ -0,0 +1,264 @@
+
+| Name        | Mandatory | Location                 | Type     | Description |
+| :---------- | :-------: | :----------------------- | :------- | :---------- |
+| *subject*   | yes       | *properties*             | *string* | MUST be set to `add`. |
+
+The request message MUST contain a tenant payload as defined in the [Tenant Format]({{< relref "#tenant-format" >}}) section below.
+
+In the payload it is also possible to put additional custom meta information about the tenant (see the [Tenant Format]({{< relref "#tenant-format" >}}) section for further information).
+ 
+**Response Message Format**
+
+A response to an *add tenant* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *201* | Created, the tenant has been successfully created. |
+| *409* | Conflict, there already exists a tenant for the given *tenant-id*. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Get Tenant Information
+
+Clients use this command to *retrieve* information about a tenant.
+
+This operation is *mandatory* to implement.
+
+**Message Flow**
+
+The following sequence diagram illustrates the flow of messages involved in a *Client* retrieving tenant information.
+
+![Get Tenant Information message flow](../tenant_GetTenantSuccess.png)
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to get tenant information in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name        | Mandatory | Location                 | Type     | Description |
+| :---------- | :-------: | :----------------------- | :------- | :---------- |
+| *subject*   | yes       | *properties*             | *string* | MUST be set to `get`. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+**Response Message Format**
+
+A response to a *get tenant information* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message includes payload as defined in the [Tenant Format]({{< relref "#tenant-format" >}}) section below.
+
+The response message's *status* property may contain the following codes:
+
+| Code | Description |
+| :--- | :---------- |
+| *200* | OK, the payload contains the tenant information for the requested tenant. |
+| *404* | Not Found, there is no tenant identified with the given *tenant-id*. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Update Tenant
+
```

Comment:
As written below for the remove operation, I have concerns here as well:

update may affect the overall system behaviour, e.g.:
- an update sets the requirement of authentication for devices, which was not set before
- an update limits the amount of traffic (that was bigger before)
and so on.

While  this is "only" an API, it still makes sense IMHO to add some lines about possible impacts for a cconcrete implementation. How these points are finally implemented, can be left open.

Reply:
I added some more description about the possible effects.